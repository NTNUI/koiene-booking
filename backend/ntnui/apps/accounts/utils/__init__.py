from accounts.utils.mail_utils import send_gsuite_confirmation_mail
