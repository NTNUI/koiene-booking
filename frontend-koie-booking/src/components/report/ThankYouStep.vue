<template>
  <v-layout :class="$style.container" :dark="true">
    <h1 :class="$style.heading">{{ $t('report.stepThankYouTitle') }}</h1>
    <v-layout :class="$style.separator">
      <v-layout class="px-4">
        <p>{{ $t('report.stepThankYouDescription') }}</p>
      </v-layout>
    </v-layout>
  </v-layout>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'ReportThankYouStep',
});
</script>

<style lang="scss" module>
.container {
  display: flex;
  flex-direction: column;
}
.heading {
  margin-top: 20px;
  align-self: center;
}
.separator {
  border-left: solid 4px;
  border-color: #2f3a50;
  display: flex;
  flex-direction: column;
  margin-top: 24px;
}
.separator > h3 {
  padding: 16px;
}
</style>
