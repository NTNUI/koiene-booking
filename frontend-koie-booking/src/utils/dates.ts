import dayjs from 'dayjs';
require('dayjs/locale/nb');
dayjs.locale('nb');

export function getDateString(baseDate?: string, addDays?: number): string {
  let resultString: string;
  if (baseDate && addDays) {
    resultString = new Date(new Date(baseDate).getTime() + addDays * 24 * 60 * 60 * 1000).toISOString().substr(0, 10);
  } else if (baseDate) {
    resultString = new Date(baseDate).toISOString().substr(0, 10);
  } else if (addDays) {
    resultString = new Date(new Date().getTime() + addDays * 24 * 60 * 60 * 1000).toISOString().substr(0, 10);
  } else {
    resultString = new Date().toISOString().substr(0, 10);
  }
  return resultString;
}

export function getDate(baseDate?: string, addDays?: number): Date {
  let resultDate: Date;
  if (baseDate && addDays) {
    resultDate = new Date(new Date(baseDate).getTime() + addDays * 24 * 60 * 60 * 1000);
  } else if (baseDate) {
    resultDate = new Date(baseDate);
  } else if (addDays) {
    resultDate = new Date(new Date().getTime() + addDays * 24 * 60 * 60 * 1000);
  } else {
    resultDate = new Date();
  }
  return resultDate;
}

export function addToDate(startDate: string, howMany: number, timeUnit: dayjs.OpUnitType): string {
  const date = dayjs(startDate + 'T00:00:00.000Z');
  return date.add(howMany, timeUnit).format('YYYY-MM-DD');
}

export function formatDate(dateISO: string, formatString: string): string {
  const date = dayjs(dateISO + 'T00:00:00.000Z');
  return date.format(formatString);
}
