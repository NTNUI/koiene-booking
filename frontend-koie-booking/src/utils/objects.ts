export const clone = (object: any) => Object.assign({}, object);

export const renameKey = (object: any, key: string, newKey: string): any => {
  const value = object[key];
  delete object[key];
  object[newKey] = value;
  return object;
};

export const sortObjectByKey = (unordered: any) => {
  let ordered: any;
  ordered = {};
  Object.keys(unordered)
    .sort()
    .forEach(function(key) {
      ordered[key] = unordered[key];
    });
  return ordered;
};
