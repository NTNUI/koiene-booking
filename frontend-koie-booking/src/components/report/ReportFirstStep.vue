<template>
  <v-layout :class="$style.container" :dark="true">
    <h1 :class="$style.heading">{{ $t('report.reportTitle') }}</h1>
    <v-layout :class="$style.separator">
      <v-layout class="px-4">
        <p>{{ $t('report.reportDescription') }}</p>
      </v-layout>
      <v-layout class="px-4">
        <p>
          {{ $t('report.reportYourTripTo') }}
          <span :class="$style.capitalize">{{ koie }}</span>
          {{ $t('report.reportFrom') }}
          {{ fromDate }}
          {{ $t('report.reportTo') }}
          {{ toDate }}.
        </p>
      </v-layout>
    </v-layout>
  </v-layout>
</template>

<script lang="ts">
import Vue from 'vue';
import { formatDate } from '@/utils/dates';

export default Vue.extend({
  name: 'ReportFirstStep',
  computed: {
    koie(): string {
      return this.$store.state.report.bookingInfo.koie;
    },
    fromDate(): string {
      return formatDate(this.$store.state.report.bookingInfo.from_date, 'DD.MM.YYYY');
    },
    toDate(): string {
      return formatDate(this.$store.state.report.bookingInfo.to_date, 'DD.MM.YYYY');
    },
  },
});
</script>

<style lang="scss" module>
.container {
  display: flex;
  flex-direction: column;
}
.heading {
  margin-top: 20px;
  align-self: center;
}
.separator {
  border-left: solid 4px;
  border-color: #2f3a50;
  display: flex;
  flex-direction: column;
  margin-top: 24px;
}
.separator > h3 {
  padding: 16px;
}
.capitalize {
  text-transform: capitalize;
}
</style>
