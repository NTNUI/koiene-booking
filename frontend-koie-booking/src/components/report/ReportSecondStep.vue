<template>
  <v-layout :class="$style.container" :dark="true">
    <h1 :class="$style.heading">{{ $t('report.step2') }}</h1>
    <v-form v-model="validForm" :class="$style.form">
      <v-layout :class="$style.separator">
        <h3 class="py-4" :class="$style.supplyRegistration">{{ $t('report.gass_is_full') }}</h3>
        <v-layout class="px-4">
          <v-radio-group v-model="gasIsFull" mandatory required :color="$scssVars.globalColorBackgroundLight">
            <v-radio :value="true" :label="$t('report.gass_full')"></v-radio>
            <v-radio :value="false" :label="$t('report.gass_empty')"></v-radio>
          </v-radio-group>
        </v-layout>
      </v-layout>
      <v-layout :class="$style.separator">
        <h3 class="py-4" :class="$style.form">{{ $t('report.firewood_supply') }}</h3>
        <v-layout class="px-4">
          <v-slider
            v-model="firewoodSupply"
            required
            :tick-labels="firewoodSupplyLabels"
            :max="4"
            step="1"
            ticks="always"
            tick-size="5"
          ></v-slider>
        </v-layout>
      </v-layout>
      <v-layout :class="$style.separator">
        <h3 class="py-4" :class="$style.form">{{ $t('report.chopped_up_wood_supply') }}</h3>
        <v-layout class="px-4">
          <v-slider
            v-model="choppedUpWoodSupply"
            required
            :tick-labels="choppedUpSupplyLabels"
            :max="4"
            step="1"
            ticks="always"
            tick-size="5"
          ></v-slider>
        </v-layout>
      </v-layout>
    </v-form>
  </v-layout>
</template>

<script lang="ts">
import Vue from 'vue';
import { ReportSecondStepData } from '../../types/report';

export default Vue.extend({
  name: 'ReportSecondStep',
  data(): ReportSecondStepData {
    return {
      validForm: true,
      firewoodSupplyLabels: [
        `${this.$t('report.firewood_supply_empty')}`,
        '',
        '',
        '',
        `${this.$t('report.firewood_supply_full')}`,
      ],
      choppedUpSupplyLabels: [
        `${this.$t('report.chopped_up_wood_supply_empty')}`,
        '',
        '',
        '',
        `${this.$t('report.chopped_up_wood_supply_full')}`,
      ],
    };
  },
  computed: {
    gasIsFull: {
      get() {
        return this.$store.state.report.reportData.gas_is_full;
      },
      set(value: boolean) {
        this.$store.commit('report/setGasIsFull', value);
        this.$store.commit('report/setEdited', true);
      },
    },
    firewoodSupply: {
      get() {
        return this.$store.state.report.reportData.firewood;
      },
      set(value: number) {
        this.$store.commit('report/setFirewoodSupply', value);
        this.$store.commit('report/setEdited', true);
      },
    },
    choppedUpWoodSupply: {
      get() {
        return this.$store.state.report.reportData.chopped_up_wood;
      },
      set(value: number) {
        this.$store.commit('report/setChoppedUpWoodSupply', value);
        this.$store.commit('report/setEdited', true);
      },
    },
  },
});
</script>

<style lang="scss" module>
.container {
  display: flex;
  flex-direction: column;
}
.heading {
  margin-top: 20px;
  align-self: center;
}
.separator {
  border-left: solid 4px;
  border-color: #2f3a50;
  display: flex;
  flex-direction: column;
  margin-top: 24px;
}
.separator > h3 {
  padding: 16px;
}
.form {
  padding-left: 16px;
  padding-right: 16px;
  @media only screen and (max-width: 600px) {
    padding-left: 8px;
    padding-right: 8px;
  }
}
.fieldWrapper {
  display: flex;
  width: 100%;
  align-items: center;
}
.numberField {
  padding-right: 10px;
}
</style>
