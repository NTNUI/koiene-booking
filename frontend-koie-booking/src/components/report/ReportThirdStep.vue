<template>
  <v-layout :class="$style.container" :dark="true">
    <h1 :class="$style.heading">{{ $t('report.step3') }}</h1>
    <v-form v-model="validForm" :class="$style.form">
      <v-layout :class="$style.separator">
        <h3 class="py-4" :class="$style.supplyRegistration">{{ $t('report.water_equipment.boat') }}</h3>
        <v-layout class="px-4">
          <v-radio-group v-model="boatStatus" mandatory required :color="$scssVars.globalColorBackgroundLight">
            <v-radio :value="3" :label="$t('report.water_equipment.missing_boat')"></v-radio>
            <v-radio :value="1" :label="$t('report.water_equipment.broken')"></v-radio>
            <v-radio :value="2" :label="$t('report.water_equipment.missing_paddles')"></v-radio>
            <v-radio :value="4" :label="$t('report.water_equipment.unsure')"></v-radio>
            <v-radio :value="0" :label="$t('report.water_equipment.no_faults')"></v-radio>
          </v-radio-group>
        </v-layout>
      </v-layout>
      <v-layout :class="$style.separator">
        <h3 class="py-4" :class="$style.supplyRegistration">{{ $t('report.water_equipment.canoe') }}</h3>
        <v-layout class="px-4">
          <v-radio-group v-model="canoeStatus" mandatory required :color="$scssVars.globalColorBackgroundLight">
            <v-radio :value="3" :label="$t('report.water_equipment.missing_canoe')"></v-radio>
            <v-radio :value="1" :label="$t('report.water_equipment.broken')"></v-radio>
            <v-radio :value="2" :label="$t('report.water_equipment.missing_paddles')"></v-radio>
            <v-radio :value="4" :label="$t('report.water_equipment.unsure')"></v-radio>
            <v-radio :value="0" :label="$t('report.water_equipment.no_faults')"></v-radio>
          </v-radio-group>
        </v-layout>
      </v-layout>
      <v-layout :class="$style.separator">
        <h3 class="py-4" :class="$style.supplyRegistration">{{ $t('report.water_equipment.life_jackets') }}</h3>
        <v-layout class="px-4">
          <v-radio-group v-model="lifeJacketsStatus" mandatory required :color="$scssVars.globalColorBackgroundLight">
            <v-radio :value="2" :label="$t('report.water_equipment.missing_life_jackets')"></v-radio>
            <v-radio :value="1" :label="$t('report.water_equipment.broken')"></v-radio>
            <v-radio :value="3" :label="$t('report.water_equipment.unsure')"></v-radio>
            <v-radio :value="0" :label="$t('report.water_equipment.no_faults')"></v-radio>
          </v-radio-group>
        </v-layout>
      </v-layout>
    </v-form>
  </v-layout>
</template>

<script lang="ts">
import Vue from 'vue';
import { ReportThirdStepData } from '../../types/report';

export default Vue.extend({
  name: 'ReportThirdStep',
  data(): ReportThirdStepData {
    return {
      validForm: true,
    };
  },
  computed: {
    boatStatus: {
      get() {
        return this.$store.state.report.reportData.boat_status;
      },
      set(value: number) {
        this.$store.commit('report/setBoatStatus', value);
        this.$store.commit('report/setEdited', true);
      },
    },
    canoeStatus: {
      get() {
        return this.$store.state.report.reportData.canoe_status;
      },
      set(value: number) {
        this.$store.commit('report/setCanoeStatus', value);
        this.$store.commit('report/setEdited', true);
      },
    },
    lifeJacketsStatus: {
      get() {
        return this.$store.state.report.reportData.life_jackets_status;
      },
      set(value: number) {
        this.$store.commit('report/setLifeJacketsStatus', value);
        this.$store.commit('report/setEdited', true);
      },
    },
  },
});
</script>

<style lang="scss" module>
.container {
  display: flex;
  flex-direction: column;
}
.heading {
  margin-top: 20px;
  align-self: center;
}
.separator {
  border-left: solid 4px;
  border-color: #2f3a50;
  display: flex;
  flex-direction: column;
  margin-top: 24px;
}
.separator > h3 {
  padding: 16px;
}
.feedbackField {
  width: 100%;
  padding-right: 10px;
}
</style>
