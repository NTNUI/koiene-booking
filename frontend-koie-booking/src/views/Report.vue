<template>
  <v-container class="global-max-width">
    <v-row>
      <v-col cols="12" sm="8" class="mx-auto" :class="$style.mainColumn">
        <v-stepper v-model="step" alt-labels :class="$style.stepper">
          <v-stepper-header :class="$style.stepperHeader">
            <v-stepper-step :complete="step > 1" :step="1">
              {{ $t('report.step1') }}
            </v-stepper-step>
            <v-divider></v-divider>
            <v-stepper-step :complete="step > 2" :step="2">
              {{ $t('report.step2') }}
            </v-stepper-step>
            <v-divider></v-divider
            ><v-stepper-step :complete="step > 3" :step="3">
              {{ $t('report.step3') }}
            </v-stepper-step>
            <v-divider></v-divider
            ><v-stepper-step :complete="step > 4" :step="4">
              {{ $t('report.step4') }}
            </v-stepper-step>
          </v-stepper-header>
          <v-layout>
            <ErrorCard v-if="apiError" />
            <LoadingSpinner v-else-if="isLoading" />
            <FirstStep v-else-if="step === 1" />
            <SecondStep v-else-if="step === 2" />
            <ThirdStep v-else-if="step === 3" />
            <FourthStep v-else-if="step === 4" />
            <ThankYouStep v-else-if="step === 5" />
          </v-layout>
        </v-stepper>
        <div v-show="!isLoading && step < 5" :class="$style.btnWrapper">
          <v-btn
            v-if="step > 1"
            :color="this.$scssVars.globalColorPrimary"
            raised="true"
            :dark="true"
            data-test="btnPrev"
            @click="prevStep(step)"
            >{{ $t('report.back_button') }}</v-btn
          >
          <v-btn
            v-if="step < steps"
            :disabled="!isValid"
            :color="this.$scssVars.globalColorPrimary"
            :class="$style.nextStepButton"
            raised="true"
            :dark="true"
            data-test="btnNext"
            @click="nextStep(step)"
            >{{ $t('report.next_button') }}</v-btn
          >
          <v-btn
            v-if="step === steps"
            :color="this.$scssVars.globalColorPrimary"
            :class="$style.nextStepButton"
            raised="true"
            :dark="true"
            data-test="btnNext"
            @click="done"
            >{{ $t('report.done_button') }}</v-btn
          >
        </div>
      </v-col>
    </v-row>
  </v-container>
</template>

<script lang="ts">
import ErrorCard from '@/components/ErrorCard.vue';
import LoadingSpinner from '@/components/LoadingSpinner.vue';
import FirstStep from '@/components/report/FirstStep.vue';
import SecondStep from '@/components/report/SecondStep.vue';
import ThirdStep from '@/components/report/ThirdStep.vue';
import FourthStep from '@/components/report/FourthStep.vue';
import ThankYouStep from '@/components/report/ThankYouStep.vue';
import { ReportData } from '../types/report';
import Vue from 'vue';

export default Vue.extend({
  name: 'Report',

  components: {
    ErrorCard,
    LoadingSpinner,
    FirstStep,
    SecondStep,
    ThirdStep,
    FourthStep,
    ThankYouStep,
  },

  data(): ReportData {
    return {
      step: 1,
      steps: 5,
    };
  },

  computed: {
    isValid(): string {
      return this.$store.state.report.validForm;
    },
    isLoading(): boolean {
      return this.$store.state.report.isLoading;
    },
    apiError(): boolean {
      return this.$store.state.report.error;
    },
  },

  mounted() {
    this.step = 1;
    this.$store.commit('report/setStep', 1);
    this.$store.commit('report/setValidForm', true);
    this.$store.commit('report/setEdited', false);
    const booking = Number(this.$route.params.booking_id);
    this.$store.commit('report/setBookingID', booking);
    this.$store.dispatch('report/FETCH_BOOKING', booking);
  },

  methods: {
    nextStep(n: number) {
      this.step = ++n % (this.steps + 1);
      this.$store.commit('report/setStep', Number(this.$store.state.report.step) + 1);
    },
    prevStep(n: number) {
      this.step = --n % (this.steps + 1);
      this.$store.commit('report/setStep', Number(this.$store.state.report.step) - 1);
      if (this.step === 1) {
        this.$store.commit('report/setValidForm', true);
      }
    },
    done() {
      this.$store.dispatch('report/CREATE_REPORT', this.$store.state.report.reportData);
    },
    resetReportInfo() {},
  },
});
</script>

<style lang="scss" module>
.mainColumn {
  @media only screen and (max-width: 600px) {
    padding-left: 0px;
    padding-right: 0px;
  }
}

.stepper {
  min-height: 80vh;
  box-shadow: none;
  background: none !important;
  @media only screen and (min-width: 1500px) {
    min-height: 60vh;
  }
}
.stepperHeader {
  box-shadow: none;
}

.btnWrapper {
  display: flex;
  align-items: center;
  justify-content: center;
  bottom: 5%;
}
.nextStepButton {
  margin-left: 20px;
}
.errorMsg {
  margin: 10px 16px 10px 16px;
}
</style>
