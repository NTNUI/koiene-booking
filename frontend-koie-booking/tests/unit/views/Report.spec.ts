import Vue from 'vue';
import Vuetify from 'vuetify';
import flushPromises from 'flush-promises';
import mockAxios from 'jest-mock-axios';

Vue.use(Vuetify);

// Utilities
import { Wrapper, ThisTypedShallowMountOptions } from '@vue/test-utils';
import { createWrapper } from '../utils';

// Components or views
import Report from '@/views/Report.vue';
import ReportFirstStep from '@/components/report/ReportFirstStep.vue';
import ReportSecondStep from '@/components/report/ReportSecondStep.vue';
import ReportThirdStep from '@/components/report/ReportThirdStep.vue';
import ReportFourthStep from '@/components/report/ReportFourthStep.vue';
import ThankYouStep from '@/components/report/ThankYouStep.vue';

describe('View Booking.vue', () => {
  let wrapper: Wrapper<any>;
  let wrapperOptions: ThisTypedShallowMountOptions<any>;

  beforeEach(() => {
    // We need this in order to solve a v-slider bug causing a
    // warning https://github.com/vuetifyjs/vuetify/issues/1210#issuecomment-319624495
    const app = document.createElement('div');
    app.setAttribute('data-app', 'true');
    document.body.appendChild(app);

    wrapperOptions = {
      mocks: {
        $route: {
          params: { booking_id: 1 },
        },
      },
    };
    wrapper = createWrapper(Report, wrapperOptions);
  });

  describe('with water equipment', () => {
    beforeEach(async () => {
      const response = {
        data: { booking: { koie: 'flÃ¥koia', arrival_date: '1970-01-01', departure_date: '1970-01-10' } },
      };
      mockAxios.mockResponse(response);
      await flushPromises();
    });

    afterEach(() => {
      mockAxios.reset();
    });

    it('Matches snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('Buttons should be hidden if report is loading', () => {
      expect(wrapper.vm.$el.querySelector('.btnWrapper').style.display).toBe('');
      wrapperOptions = {
        mocks: {
          $route: {
            params: { id: 1 },
          },
        },
        computed: {
          isLoading() {
            return true;
          },
        },
      };
      wrapper = createWrapper(Report, wrapperOptions);
      expect(wrapper.vm.$el.querySelector('.btnWrapper').style.display).toBe('none');
    });

    it('Buttons should be hidden if there is an api error', () => {
      expect(wrapper.vm.$el.querySelector('.btnWrapper').style.display).toBe('');
      wrapperOptions = {
        mocks: {
          $route: {
            params: { id: 1 },
          },
        },
        computed: {
          apiError() {
            return true;
          },
        },
      };
      wrapper = createWrapper(Report, wrapperOptions);
      expect(wrapper.vm.$el.querySelector('.btnWrapper').style.display).toBe('none');
    });

    it('Button_next renders next report step', () => {
      wrapper.find('[data-test="btnNext"]').trigger('click');

      expect(wrapper.contains(ReportSecondStep)).toBe(true);
      expect(wrapper.contains(ReportFirstStep)).toBe(false);
    });

    it('Button_prev renders prev report step', () => {
      wrapper.find('[data-test="btnNext"]').trigger('click');
      wrapper.find('[data-test="btnPrev"]').trigger('click');

      expect(wrapper.contains(ReportFirstStep)).toBe(true);
      expect(wrapper.contains(ReportSecondStep)).toBe(false);
    });

    it('Button_done renders thank you step', async () => {
      const btnNext = wrapper.find('[data-test="btnNext"]');
      btnNext.trigger('click');
      btnNext.trigger('click');
      btnNext.trigger('click');
      wrapper.find('[data-test="btnDone"]').trigger('click');

      const response = { data: {} };
      mockAxios.mockResponse(response);
      await flushPromises();

      expect(wrapper.contains(ThankYouStep)).toBe(true);
      expect(wrapper.contains(ReportFourthStep)).toBe(false);
    });
  });

  describe('without water equipment', () => {
    beforeEach(async () => {
      const response = {
        data: { booking: { koie: 'lyngli', arrival_date: '1970-01-01', departure_date: '1970-01-10' } },
      };
      mockAxios.mockResponse(response);
      await flushPromises();
    });

    afterEach(() => {
      mockAxios.reset();
    });

    it('Matches snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('Buttons should be hidden if report is loading', () => {
      expect(wrapper.vm.$el.querySelector('.btnWrapper').style.display).toBe('');
      wrapperOptions = {
        mocks: {
          $route: {
            params: { id: 1 },
          },
        },
        computed: {
          isLoading() {
            return true;
          },
        },
      };
      wrapper = createWrapper(Report, wrapperOptions);
      expect(wrapper.vm.$el.querySelector('.btnWrapper').style.display).toBe('none');
    });

    it('Buttons should be hidden if there is an api error', () => {
      expect(wrapper.vm.$el.querySelector('.btnWrapper').style.display).toBe('');
      wrapperOptions = {
        mocks: {
          $route: {
            params: { id: 1 },
          },
        },
        computed: {
          apiError() {
            return true;
          },
        },
      };
      wrapper = createWrapper(Report, wrapperOptions);
      expect(wrapper.vm.$el.querySelector('.btnWrapper').style.display).toBe('none');
    });

    it('Button_next renders next report step', () => {
      wrapper.find('[data-test="btnNext"]').trigger('click');

      expect(wrapper.contains(ReportSecondStep)).toBe(true);
      expect(wrapper.contains(ReportFirstStep)).toBe(false);
    });

    it('Button_prev renders prev report step', () => {
      wrapper.find('[data-test="btnNext"]').trigger('click');
      wrapper.find('[data-test="btnPrev"]').trigger('click');

      expect(wrapper.contains(ReportFirstStep)).toBe(true);
      expect(wrapper.contains(ReportSecondStep)).toBe(false);
    });

    it('Button_done renders thank you step', async () => {
      const btnNext = wrapper.find('[data-test="btnNext"]');
      btnNext.trigger('click');
      btnNext.trigger('click');
      wrapper.find('[data-test="btnDone"]').trigger('click');

      const response = { data: {} };
      mockAxios.mockResponse(response);
      await flushPromises();

      expect(wrapper.contains(ThankYouStep)).toBe(true);
      expect(wrapper.contains(ReportThirdStep)).toBe(false);
    });
  });
});
