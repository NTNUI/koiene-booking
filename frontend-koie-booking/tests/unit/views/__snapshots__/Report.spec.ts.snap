// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View Booking.vue with water equipment Matches snapshot 1`] = `
<div class="container global-max-width">
  <div class="row">
    <div class="mx-auto col-sm-8 col-12 mainColumn">
      <div class="v-stepper v-stepper--alt-labels theme--light stepper">
        <div class="v-stepper__header stepperHeader">
          <div class="v-stepper__step v-stepper__step--active"><span class="v-stepper__step__step primary">1</span>
            <div class="v-stepper__label">
              Info
            </div>
          </div>
          <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
          <div class="v-stepper__step v-stepper__step--inactive"><span class="v-stepper__step__step">2</span>
            <div class="v-stepper__label">
              Supplies
            </div>
          </div>
          <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
          <div class="v-stepper__step v-stepper__step--inactive"><span class="v-stepper__step__step">3</span>
            <div class="v-stepper__label">
              Boat and canoe
            </div>
          </div>
          <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
          <div class="v-stepper__step v-stepper__step--inactive"><span class="v-stepper__step__step">4</span>
            <div class="v-stepper__label">
              Equipment
            </div>
          </div>
        </div>
        <div class="layout">
          <div class="layout dark container">
            <h1 class="heading">Digital Cabin Report</h1>
            <div class="layout separator">
              <div class="layout px-4">
                <p>The cabin group is run purely on volunteer work and we rely on your feedback to be able to run the cabins.</p>
              </div>
              <div class="layout px-4">
                <p>
                  This report concerns your trip to
                  <span class="capitalize">fl√•koia</span>
                  from
                  01.01.1970
                  to
                  10.01.1970.
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="btnWrapper" style="">
        <!----> <button type="button" class="v-btn v-btn--contained theme--dark v-size--default nextStepButton" raised="true" data-test="btnNext"><span class="v-btn__content">next</span></button>
        <!---->
      </div>
    </div>
  </div>
</div>
`;

exports[`View Booking.vue without water equipment Matches snapshot 1`] = `
<div class="container global-max-width">
  <div class="row">
    <div class="mx-auto col-sm-8 col-12 mainColumn">
      <div class="v-stepper v-stepper--alt-labels theme--light stepper">
        <div class="v-stepper__header stepperHeader">
          <div class="v-stepper__step v-stepper__step--active"><span class="v-stepper__step__step primary">1</span>
            <div class="v-stepper__label">
              Info
            </div>
          </div>
          <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
          <div class="v-stepper__step v-stepper__step--inactive"><span class="v-stepper__step__step">2</span>
            <div class="v-stepper__label">
              Supplies
            </div>
          </div>
          <!---->
          <!---->
          <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
          <div class="v-stepper__step v-stepper__step--inactive"><span class="v-stepper__step__step">3</span>
            <div class="v-stepper__label">
              Equipment
            </div>
          </div>
        </div>
        <div class="layout">
          <div class="layout dark container">
            <h1 class="heading">Digital Cabin Report</h1>
            <div class="layout separator">
              <div class="layout px-4">
                <p>The cabin group is run purely on volunteer work and we rely on your feedback to be able to run the cabins.</p>
              </div>
              <div class="layout px-4">
                <p>
                  This report concerns your trip to
                  <span class="capitalize">lyngli</span>
                  from
                  01.01.1970
                  to
                  10.01.1970.
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="btnWrapper" style="">
        <!----> <button type="button" class="v-btn v-btn--contained theme--dark v-size--default nextStepButton" raised="true" data-test="btnNext"><span class="v-btn__content">next</span></button>
        <!---->
      </div>
    </div>
  </div>
</div>
`;
